api/client/connection.go

```diff
@@ -83,21 +97,30 @@ type Response struct {
 }
 
 func (c *Connection) Do(req *Request) (*Response, error) {
-	httpreq, err := req.HTTPRequest()
-	if err != nil {
-		return nil, err
-	}
-	res, err := c.client.Do(httpreq)
-	if err != nil {
-		return nil, err
-	}
-	if res.StatusCode < 200 || res.StatusCode > 299 {
-		err = parseError(res)
+	for i := 0; ; i++ {
+		httpreq, err := req.HTTPRequest()
+		if err != nil {
+			return nil, err
+		}
+		res, err := c.client.Do(httpreq)
+		if err != nil {
+			return nil, err
+		}
+		if res.StatusCode < 200 || res.StatusCode > 299 {
+			err = parseError(res)
+			if i == 0 && res.StatusCode == 401 && err.(*ErrorResponse).Err.Error() == "invalid token" {
```

Comment:
Better to use `errors.As` than a type assertion. And we as well use `api.Error`.
```suggestion
			var apierr *api.Error
			if i == 0 && res.StatusCode == 401 && errors.As(err, &apierr) && apierr.Message == "invalid token" {
```

Reply:
Can't use api.Error since for some reason this endpoint is returning an api.Error, I'll change it to ErrorResponse.